src/daft-scan/src/scan_task_iters.rs

```diff
@@ -111,3 +116,109 @@ impl Iterator for MergeByFileSize {
         }
     }
 }
+
+pub fn split_by_row_groups(
+    scan_tasks: BoxScanTaskIter,
+    max_tasks: usize,
+    split_threshold_bytes: usize,
+    min_size_bytes: usize,
+) -> BoxScanTaskIter {
+    let mut scan_tasks = itertools::peek_nth(scan_tasks);
+
+    // only split if we have a small amount of files
+    if scan_tasks.peek_nth(max_tasks).is_some() {
+        return Box::new(scan_tasks);
```

Comment:
Oh you're right. This was probably a relic of some other thing I was doing in the past that I forgot to change

Reply:
The compiler does complain when I remove the Box::new since I'm returning a PeekNth instead of a Box. I think I could possibly clone `scan_tasks` for `itertools::peek_nth` and then return the original here but this eliminates some amount of recomputation.

Reply:
Ah I see, that makes sense, thanks!

>I think I could possibly clone scan_tasks for itertools::peek_nth and then return the original here but this eliminates some amount of recomputation.

Yeah I think what you have here, which defers any cloning of the iterator until we determine that we want to short-circuit, is better than always cloning the iterator eagerly.

Ideally, just as `itertools::peak_nth()` moves the iterator into a `PeekNth` by taking ownership, we could move the iterator back into the return without any re-boxing and tell the compiler that the `PeekNth<OurIteratorType>` is itself `OurIteratorType`. We might be able to accomplish this by changing the input and output types for the splitting and merging functions in this module to `impl Iterator<Item = DaftResult<ScanTaskRef>>`, where `merge_by_sizes` can still be called with a `Box<dyn Iterator>` input but will now return `MergeFileBySizes` instead of `Box<MergeFileBySizes>`, and `split_by_row_groups` can be called with `MergeFileBySizes` input and will return the `PeekNth` instead of `Box<PeekNth>`. I _think_ this should work, but I might be missing some reason we can't make this static rather than dynamic.

Reply:
In any case, if this ends up being a headache, we can always keep the extra boxing/clone for now!